Code/BuildSystem/CMake/ezUtilsDetect.cmake

```diff
@@ -279,223 +284,150 @@ function(ez_detect_compiler)
 	set_property(GLOBAL PROPERTY EZ_CMAKE_COMPILER_MSVC OFF)
 	set_property(GLOBAL PROPERTY EZ_CMAKE_COMPILER_MSVC_140 OFF)
 	set_property(GLOBAL PROPERTY EZ_CMAKE_COMPILER_MSVC_141 OFF)
+	set_property(GLOBAL PROPERTY EZ_CMAKE_COMPILER_MSVC_142 OFF)
 	set_property(GLOBAL PROPERTY EZ_CMAKE_COMPILER_CLANG OFF)
 	set_property(GLOBAL PROPERTY EZ_CMAKE_COMPILER_GCC OFF)
-
 	
-	if (GENERATOR_MSVC) # Visual Studio Compiler
-	  
-	  message (STATUS "Compiler is MSVC (EZ_CMAKE_COMPILER_MSVC) version ${MSVC_VERSION}")
-
+	set(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")
+	try_compile(COMPILE_RESULT
+		${CMAKE_CURRENT_BINARY_DIR}
+		${CMAKE_SOURCE_DIR}/${EZ_SUBMODULE_PREFIX_PATH}/Code/BuildSystem/ProbingSrc/ArchitectureDetect.c
+		OUTPUT_VARIABLE COMPILE_OUTPUT
+	)
+	if(NOT COMPILE_RESULT)
+		message(FATAL_ERROR "Failed to detect compiler / target architecture. Compiler output: ${COMPILE_OUTPUT}")
+	endif()
+	
+	if(${COMPILE_OUTPUT} MATCHES "ARCH:'([^']*)'")
+		set(EZ_DETECTED_ARCH ${CMAKE_MATCH_1})
+	else()
+		message(FATAL_ERROR "The compile test did not output the architecture. Compiler broken? Compiler output: ${COMPILE_OUTPUT}")
+	endif()
+	
+	if(${COMPILE_OUTPUT} MATCHES "COMPILER:'([^']*)'")
+		set(EZ_DETECTED_COMPILER ${CMAKE_MATCH_1})
+	else()
+		message(FATAL_ERROR "The compile test did not output the compiler. Compiler broken? Compiler output: ${COMPILE_OUTPUT}")
+	endif()
+	
+	if(EZ_DETECTED_COMPILER STREQUAL "msvc")
+		if(${COMPILE_OUTPUT} MATCHES "MSC_VER:'([^']*)'")
+			set(EZ_DETECTED_MSVC_VER ${CMAKE_MATCH_1})
+		else()
+			message(FATAL_ERROR "The compile test did not output the MSC_VER. Compiler broken? Compiler output: ${COMPILE_OUTPUT}")
+		endif()
+	endif()
+	
+	if(EZ_DETECTED_COMPILER STREQUAL "msvc") # Visual Studio Compiler
+	  message (STATUS "Compiler is MSVC (EZ_CMAKE_COMPILER_MSVC) version ${EZ_DETECTED_MSVC_VER}")
+		
 	  set_property(GLOBAL PROPERTY EZ_CMAKE_COMPILER_MSVC ON)
 
-	  if(MSVC_VERSION GREATER_EQUAL 1910)
+      if(EZ_DETECTED_MSVC_VER GREATER_EQUAL 1920)
+	  
+		message (STATUS "Compiler is Visual Studio 2019 (EZ_CMAKE_COMPILER_MSVC_142)")
+		set_property(GLOBAL PROPERTY EZ_CMAKE_COMPILER_MSVC_142 ON)
+		set_property(GLOBAL PROPERTY EZ_CMAKE_COMPILER_POSTFIX "2019")
```

Comment:
This is great. Can you also have a look at ez_detect_project_name in ezUtilsDetect.cmake, I think it doesn't work as desired on VS 2019 (it still seems to name the solution "2017" IIRC).

Something similar happens with the binary output path, which seems to be stuck to be "WinVs2017Debug64" even for VS 2019. See #223. Would be great if you could test whether your change fixes those problems.

Reply:
ez_detect_project name is already correct